server/src/main/java/org/candlepin/pinsetter/core/PinsetterTriggerListener.java

```diff
@@ -64,5 +70,23 @@ public void triggerMisfired(Trigger trigger) {
             trigger.getNextFireTime(),
             trigger.getMisfireInstruction(),
             trigger.getPriority());
+
+        String id = trigger.getJobKey().getName();
+        JobStatus js = jobCurator.find(id);
+        if (js == null) {
+            throw new RuntimeException("No JobStatus for job with id of " + id);
+        }
+        // if may not refire again, change to fail
+        if (!trigger.mayFireAgain()) {
+            log.warn("Job {} not allowed to fire again. Setting state to FAILED.", id);
+            js.setState(JobStatus.JobState.FAILED);
+            js.setResult("Failed run. Will not attempt again.");
+        }
+        else {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+            js.setResult("Will reattempt job at or after " + sdf.format(trigger.getNextFireTime()));
+            js.setState(JobStatus.JobState.PENDING);
+        }
+        jobCurator.merge(js);
```

Comment:
Is there any code in candlepin that might attempt to restart a job that is in PENDING state? Since this is a misfire, and further scheduling will be handled by Quartz itself, let's make sure that candlepin won't attempt to restart this itself.